element-hq/element-x-android
# FFs can now be toggled in release builds too

## Type of change


## Content

- Removed `StaticFeatureFlagProvider`.
- Always provide `PreferencesFeatureFlagProvider`.
- For the default values of feature flags, use a lambda with a `BuildMeta` parameter so we can customize the return value based on its data.

## Motivation and context

Fixes #3073.

## Tests



- In a release build, access the hidden developer settings.
- Toggle one of the many options.
- Check the change took effect.

## Tested devices

- OS version(s): 14

## Checklist
